src/Microsoft.AspNet.Authentication.Facebook/FacebookAuthenticationMiddleware.cs

```diff
@@ -37,11 +37,11 @@ public class FacebookAuthenticationMiddleware : OAuthAuthenticationMiddleware<Fa
         {
             if (string.IsNullOrWhiteSpace(Options.AppId))
             {
-                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources.Exception_OptionMustBeProvided, nameof(Options.AppId)));
+                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources.Exception_OptionMustBeProvided, nameof(Options)));
```

Comment:
@Eilon the ```Options``` parameter is there!!, I'm agree that the message won't make sense, but I think we need to modify the message, not the exception message while ```Options``` is the original source for ```AppId```